### DISCUSSION TITLE
Colorizing grains by their short axis values

### DISCUSSION THREAD

**Discussion by rrkamath** [mtex-toolbox/mtex#1200] [Ask Anything] [answered]
Hi,

I was able to access the numerical values of the short axis using fitEllipse. I wanted to go one step further and wanted to colorize the grains by the short axis values of the ellipse as done for the aspect ratio here: https://mtex-toolbox.github.io/EllipseBasedParameters.html. 

The code I used for the aspect ratio was:

```
plot(grains,grains.aspectRatio,'linewidth',2,'micronbar','off')
mtexColorbar('title','aspect ratio')
```

I tried replacing "grains.aspectRatio" with "grains.b" (b was the matrix with short axes according to above link), but it did not work. Can someone please help me with this ? Thanks !

Best
Rakesh

**Comment by zmichels** [comment, marked as answer]
Interesting. The following works fine for me to make a figure of grains and plot them by the short-axis length: % compute fit-ellipses and append grains variable with ellipse info[grains.prop.omega, grains.prop.a, grains.prop.b] = fitEllipse(grains) % plot itfigure,plot(grains,grains.b)   Can you provide some more details about the process leading up to the error? Or, can you try the above and see if it works for you? Cheers,Zach    From: rrkamathSent: Monday, September 13, 2021 3:38 PMTo: mtex-toolbox/mtexCc: SubscribedSubject: [mtex-toolbox/mtex] Colorizing grains by their short axis values (Discussion #1200) Hi,I was able to access the numerical values of the short axis using fitEllipse. I wanted to go one step further and wanted to colorize the grains by the short axis values of the ellipse as done for the aspect ratio here: https://mtex-toolbox.github.io/EllipseBasedParameters.html.The code I used for the aspect ratio was:plot(grains,grains.aspectRatio,'linewidth',2,'micronbar','off')mtexColorbar('title','aspect ratio')I tried replacing "grains.aspectRatio" with "grains.b" (b was the matrix with short axes according to above link), but it did not work. Can someone please help me with this ? Thanks !BestRakesh—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Reply by rrkamath** [reply]
Hi Zach,

Thank you for your reply. I incorporated your code into mine and it looks like this:

```
`[grains.prop.omega, grains.prop.a, grains.prop.b] = grains.fitEllipse;

% visualize the short axis
figure
plot(parentGrains('Ti-bcc'),parentGrains('Ti-bcc').prop.b,'linewidth',2,'micronbar','off')
mtexColorbar('title','short axis')

% quantify the short axis
figure('Name','Short Axis - Number Distribution')
histogram(parentGrains('Ti-bcc').prop.b)
xlabel('short axis')
ylabel('number of grains')`
```

The error I am getting is:

> Reference to non-existent field 'b'.

> Error in grain2d/subsref (line 44)
> [varargout{1:nargout}] = builtin('subsref',grains,s);

> Error in Ti64_P12 (line 334)
> plot(parentGrains('Ti-bcc'),parentGrains('Ti-bcc').prop.b,'linewidth',2,'micronbar','off')

Do you know how I can solve this ?

Rakesh

**Comment by zmichels** [comment]
I imagine the trouble is that the example I sent uses a grainset in a variable named “grains”, whereas everything you added refers to a variable named parentGrains… and the computed axes lengths were not assigned to that variable. You could try replacing the “grains” pets of my code snippet with “parentGrains” and see if that works maybe. Let me know.

Cheers,
Zach

**Reply by rrkamath** [reply]
Oh shoot. Silly mistake on my part. I changed the code and it worked. Thanks, Zach.

Cheers,
Rakesh

**Comment by zmichels** [comment]
Great!

Cheers,
Zach
